src/fuzz/fuzz-hidraw-id-desc.c

```diff
@@ -0,0 +1,20 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "hidraw_id/hidraw_id_desc.h"
+#include "fuzz.h"
+#include "log.h"
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+        struct HidrawIdDescResult res;
+        /* We don't want to fill the logs with messages about parse errors.
+         * Disable most logging if not running standalone */
+        if (!getenv("SYSTEMD_LOG_LEVEL"))
+                log_set_max_level(LOG_CRIT);
+
+        parse_hid_report_desc(data, size, &res);
```

Comment:
Generally we use `(void)` if we ignore the result of a function to silence static analysers but in this particular case I think it would be better to process `res` mostly to help MSan to make sure it's initialized properly after `parse_hid_report_desc` is finished.

Reply:
Could you verify that I simulate this processing in the correct way? I am storing a value depending on everything in res in a volatile non-local variable, which I do in order to get around both compiler optimizations and warnings about unused variables.

Reply:
@FabianHenneke MSan isn't triggered when uninitialized memory is copied directly without using anywhere so the assignment is unlikely to work even though the variable doesn't seem to be compiled out. I think we should start using `__msan_check_mem_is_initialized` at some point but to be honest it's the rabbit hole I'd like to go down later. I think the original version with `(void)` before `parse_hid_report_desc` should be good enough for now.

Reply:
I went through our fuzzers and it turns out we almost never try to make sure that whatever we get after parsing data is properly initialized and most likely miss bugs that can easily 
 be caught by MSan. I'll add it to the fuzzing TODO list right after i386 support I started to play with in https://github.com/systemd/systemd/pull/12608.

I think it might be a good idea to just get back to `(void) parse_hid_report_desc()`. Sorry about the misdirection.

Reply:
No worries, thank you for helping out with the fuzzer results. I'm open to explore the use of `__msan_check_mem_is_initialized` once it reaches the top of that list.